### PR TITLE
Nemo update

### PR DESCRIPTION
**Pull request by lupesko**
This change updates the nemo example as follows:
- Fixed a bug in the seq command which launched one process too many, failing the job
- Update to run_name to have an informative default
- Updating base image to latest nemo image 22.11
- Added comments better document the setup
- Moved apt install into the integrations section
- Updated a few hyper parameters to ensure successful run

Both YAMLs were tested and ran successfully on r7z2 (tested with @bcui19 's pre-processed data stored at s3://mosaicml-internal-checkpoints-shared/brandon/NeMo/)

### CODE REVIEW

File: examples/nemo/single_node.yaml [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
 
-  # getting the vocab, merge files for the tokenizer
+command: | 
+  # Getting the tokenizer vocab and merge files
   wget https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-vocab.json
   wget https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-merges.txt
 
+  # Make sure to prepare and download the training data, as defined in NeMo documentation: https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/nlp/nemo_megatron/gpt/gpt_training.html
+
+  # Make sure to update the training datset path below
```
Comment (dakinggg) [inline examples/nemo/single_node.yaml:20]:
```suggestion
  # Make sure to update the training dataset path below
```

File: examples/nemo/multi_node.yaml [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
   # getting the vocab, merge files for the tokenizer
   wget https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-vocab.json
   wget https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-merges.txt
 
-  apt update
-  apt install -y parallel
+  # Make sure to prepare and download the training data, as defined in NeMo documentation: https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/nlp/nemo_megatron/gpt/gpt_training.html
 
-  seq 0 8 | parallel -u \ 'CUDA_VISIBLE_DEVICES={} RANK=$(( $NODE_RANK * 8 + {} )) python3 examples/nlp/language_modeling/megatron_gpt_pretraining.py \
+  # Make sure to update the training datset path below
```
Comment (dakinggg) [inline examples/nemo/multi_node.yaml:26]:
```suggestion
  # Make sure to update the training dataset path below
```

**Review by dakinggg — COMMENTED** [review]
Will leave approval for @bcui19

File: examples/nemo/multi_node.yaml [hunk 3]
```diff
@@ -1,24 +1,39 @@
...
   --config-path=/workspace/nemo/examples/nlp/language_modeling/conf/ \
   --config-name=megatron_gpt_config.yaml \
   model.data.data_prefix=[1.0,/your_dataset_path_here/] \
   model.tokenizer.vocab_file=gpt2-vocab.json \
   model.tokenizer.merge_file=gpt2-merges.txt \
   model.optim.name="fused_adam" \
-  trainer.devices=1'
\ No newline at end of file
+  trainer.devices=1 \
+  model.micro_batch_size=32 \
```
[the remark this answers is not in the document]
Comment (lupesko) [reply examples/nemo/multi_node.yaml:35]: Good point.
@bcui19 - any suggestion on this?

[the remark this answers is not in the document]
Comment (bcui19) [reply examples/nemo/multi_node.yaml:35]: This was just for testing purposes. Part of how Nemo Megatron processes data is very specific parameters (e.g. it pre-processes by seq len, batch size I think, and a few others). Basically these parameters were intended for testing, ensuring NM worked and shouldn't be defaults.

**Comment by bcui19** [comment]
Closing because updated here:

https://github.com/mosaicml/examples/pull/138

**Pull request closed without merging**